rerun_py/rerun_sdk/rerun/components/line_strip2d_ext.py

```diff
@@ -42,13 +43,36 @@ def native_to_pa_array_override(data: LineStrip2DArrayLike, data_type: pa.DataTy
         elif isinstance(data, Sequence):
             if len(data) == 0:
                 inners = []
-            elif isinstance(data[0], np.ndarray):
-                inners = [Vec2DBatch(datum).as_arrow_array().storage for datum in data]  # type: ignore[arg-type]
-            elif isinstance(data[0], LineStrip2D):
-                inners = [Vec2DBatch(datum.points).as_arrow_array().storage for datum in data]  # type: ignore[union-attr]
             else:
-                inners = [Vec2DBatch(datum).as_arrow_array().storage for datum in data]  # type: ignore[arg-type]
-
+                # Is it a single strip or several?
+                # It could be a sequence of the style `[[0, 0], [1, 1]]` which is a single strip.
+                if all(
+                    isinstance(elem, Sequence) and len(elem) > 0 and isinstance(elem[0], numbers.Number)
+                    for elem in data
+                ):
+                    if all(len(elem) == 2 for elem in data):  # type: ignore[arg-type]
```

Comment:
These `all` filters make me nervous -- this is basically touching every point in the list. I think we can do something like check the first value to get a hint as to the shape and then try to do the right thing and handle with a value error if it fails.

Reply:
squinted at it and concluded that Vec2D/3DBatch should give a good enough error if it encounters something weird. So I removed the `all` expressions in the simplest way possible [275bdc5](https://github.com/rerun-io/rerun/pull/3668/commits/275bdc5c1cb7d8d88b202bcd73b6b73751cf155c)